src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.jdbc.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.service.ManufacturerService;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    private ManufacturerDao manufacturerDao;
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return manufacturerDao.getAll();
+    }
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return manufacturerDao.create(manufacturer);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return manufacturerDao.get(id);
```

Comment:
```suggestion
        return manufacturerDao.get(id).orElseThrow(); + exception with message 
```

Reply:
I think I don't need it because I return optional

Reply:
You can read it in CM:
get() methods in Dao layer should return Optional (not in services). In the service layer better call method get() on Optional and return the object.

Reply:
ok as you wish